packages/react-instantsearch-core-next/src/useIndex.ts

```diff
@@ -0,0 +1,23 @@
+import index from 'instantsearch.js/cjs/widgets/index/index';
+// eslint-disable-next-line import/named
+import { IndexWidgetParams } from 'instantsearch.js/es/widgets/index/index';
+import { useEffect, useMemo } from 'react';
+
+import { useIndexContext } from './useIndexContext';
+
+export type UseIndexProps = IndexWidgetParams;
+
+export function useIndex(props: UseIndexProps) {
+  const searchIndex = useIndexContext();
+  const widget = useMemo(() => index(props), [props]);
```

Comment:
We'll take care of using stable values to avoid infinite loops in another PR.

Reply:
```suggestion
  const indexWidget = useMemo(() => index(props), [props]);
```